confirm.h

```diff
@@ -27,8 +29,19 @@ struct confirm_query {/*{{{*/
 		};
 	};
 
-	uint16_t icmpid; /* icmpid == 0 fixes reverse flow id =revflow */
-			 /* uint16_t icmpseq used to identify probes */
+	union {
+		struct {
+			uint16_t src_port;
+			uint16_t dst_port;
+			uint32_t ack_number;
+			uint8_t  control_flags;
+			uint16_t window;
+			uint16_t urgent_pointer;
+		} tcp;
+
+		uint16_t icmpid; /* icmpid == 0 fixes reverse flow id =revflow */
```

Comment:
It seems this is not the same as what we are doing for TCP. For TCP, we accept values for each TCP field separately. For ICMP, we accept the flow ID and the ICMP ID.  This is weird.  Can we change it so ICMP also receives each field separately?  We should then move the comments explaining that `flowid` is stored in the forward ICMP checksum.